client/src/app/auth/pages/login/login.component.ts

```diff
@@ -1,15 +1,47 @@
 import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, FormGroupDirective, NgForm, Validators } from '@angular/forms';
+import { ErrorStateMatcher } from '@angular/material/core';
+
+/** Error when invalid control is dirty, touched, or submitted. */
+export class MyErrorStateMatcher implements ErrorStateMatcher {
```

Comment:
The error state matcher is a good candidate to be in shared folder. I propose to have a folder `/shared/errors/` inside we can create different `ErrorStateMatcher` if needed.

Reply:
I agree. I forgot that. Usually what I do is a service for forms which has functionalities like this one. I'll do that tomorrow.

Reply:
Done.